### PR TITLE
Implement AntSpecies composite class

### PR DESCRIPTION
**Pull request by Plat251**
Also includes a YAML Parser and a Trait system!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -12,6 +12,26 @@
         <maven.compiler.target>1.8</maven.compiler.target>
     </properties>
```
Comment (root-ansh) [inline pom.xml]: Please explain what this is and how will it be used in the  project?

File: src/main/java/com/worldwarofants/game/logic/AntSpecies.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    String name;
+    String nameLatin;
+    String description;
+    int health;
+    int attack;
+    int defense;
+
+    ArrayList<Trait> traits = new ArrayList<>();
+
+    public AntSpecies(String species) {
```
Comment (root-ansh) [inline src/main/java/com/worldwarofants/game/logic/AntSpecies.java]: Documentation needed (not necessarily from your side, could be taken as a task from some member. just leaving it out here if you can drop a `//todo` in it or something)

File: src/main/java/com/worldwarofants/game/logic/Trait.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package com.worldwarofants.game.logic;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.worldwarofants.game.logic.AntSpecies;
+
```
Comment (root-ansh) [inline src/main/java/com/worldwarofants/game/logic/Trait.java]: documentation needed(similar as point above) (although is it a good practice to have too much documentation?)

File: pom.xml (hunk 1 above)
Comment (Plat251) [reply pom.xml]: YAML Parsing library and a testing framework.
Many other PRs are adding it as well.
Everyone wouldn't have to add it if the Pull Requests were processed in a reasonable amount of time.

**Pull request closed without merging**